bin/node/src/service.rs

```diff
@@ -64,6 +64,22 @@ impl<N: BaseArithmetic> BackoffAuthoringBlocksStrategy<N> for LimitNonfinalized
     }
 }
 
+/// This will not be needed after A0-2984 is done
+fn revert_unfinalized_blocks(client: &FullClient) {
+    let last_finalized = client.info().finalized_number;
+    let best_block = client.info().best_number;
+    // 20 so there are no problems with finality session
+    const REVERT_THRESHOLD: u32 = 20;
+
+    let blocks_count_to_revert = best_block.saturating_sub(last_finalized + REVERT_THRESHOLD);
+
+    debug!("Reverting {} not finalized blocks", blocks_count_to_revert);
```

Comment:
Could you make this `info`? And also add `last_finalized`, `best_block` to the log message?

Reply:
Also, let's wrap this into `if blocks_count_to_revert > 0`

Reply:
done